packages/discovery-react-components/src/components/StructuredQuery/StructuredQuery.tsx

```diff
@@ -19,41 +20,49 @@ export interface StructuredQueryProps {
 
 const StructuredQuery: FC<StructuredQueryProps> = ({ messages = defaultMessages }) => {
   const mergedMessages = { ...defaultMessages, ...messages };
-  const [ruleRows, setRuleRows] = useState({ rows: [{ id: 0 }] });
-  const showAddRuleRowButton = ruleRows.rows.length < MAX_NUM_SIBLING_RULE_ROWS;
+  const [structuredQuerySelection, setStructuredQuerySelection] = useState<
+    StructuredQuerySelection
+  >({
+    groups: {
+      0: { rows: [0] }
+    }
+  });
 
-  const handleAddRuleRowOnClick = () => {
-    const newRuleRowId = ruleRows.rows[ruleRows.rows.length - 1].id + 1;
-    const newRuleRow = { id: newRuleRowId };
-    setRuleRows({
-      ...ruleRows,
-      rows: ruleRows.rows.concat(newRuleRow)
-    });
-  };
+  const showAddRuleRowButton =
+    structuredQuerySelection.groups[0].rows.length < MAX_NUM_SIBLING_RULE_ROWS;
+  const showAddRuleGroupButton =
+    keys(structuredQuerySelection.groups).length - 1 < MAX_NUM_NESTED_RULE_GROUPS;
 
   return (
     <div className={structuredQueryClass}>
-      <RuleGroupDropdown messages={mergedMessages} />
-      {ruleRows.rows.map(row => {
+      {keys(structuredQuerySelection.groups).map(id => {
```

Comment:
i'm not sure this will ensure the order of the groups unless lodash's `keys` operates differently than `Object.keys`. i was thinking that there would be a second array that stores the group ids so that we know the order of them like
```
{
  "groups": {
     "0": {
       "rows": [1]
     },
     "1": {
     }
   }
   "group_order": [0]
}
```

Reply:
Ah, ok, I see what you mean here, good point. Updated with a `group_order` 👍